scvi/module/base/_base_module.py

```diff
@@ -58,56 +63,147 @@ def __init__(
+        return self._loss_output.loss
 
     @property
     def reconstruction_loss(self) -> Union[torch.Tensor, jnp.ndarray]:  # noqa: D102
-        return self._get_dict_sum(self._reconstruction_loss)
+        return self._loss_output.dict_sum(self._loss_output.reconstruction_loss)
+
+    @property
+    def _reconstruction_loss(self):
+        return self._loss_output.reconstruction_loss
 
     @property
     def kl_local(self) -> Union[torch.Tensor, jnp.ndarray]:  # noqa: D102
-        return self._get_dict_sum(self._kl_local)
+        return self._loss_output.dict_sum(self._loss_output.kl_local)
+
+    @property
+    def _kl_local(self):
+        return self._loss_output.kl_local
+
+    @property
+    def reconstruction_loss_sum(self) -> Union[torch.Tensor, jnp.ndarray]:  # noqa: D102
+        return self._loss_output.reconstruction_loss_sum
+
+    @property
+    def kl_local_sum(self) -> Union[torch.Tensor, jnp.ndarray]:  # noqa: D102
+        return self._loss_output.kl_local_sum
+
+    @property
+    def kl_global_sum(self) -> Union[torch.Tensor, jnp.ndarray]:  # noqa: D102
+        return self._loss_output.kl_global_sum
 
     @property
     def kl_global(self) -> Union[torch.Tensor, jnp.ndarray]:  # noqa: D102
-        return self._get_dict_sum(self._kl_global)
+        return self._loss_output.dict_sum(self._loss_output.kl_global)
+
+    def dict_sum(self, x):
+        """Wrapper of LossOutput.dict_sum."""
+        return self._loss_output.dict_sum(x)
+
+
+@chex.dataclass
+class LossOutput:
+    """
+    Loss signature for models.
+
+    This class provides an organized way to record the model loss, as well as
+    the components of the ELBO. This may also be used in MLE, MAP, EM methods.
+    The loss is used for backpropagation during inference. The other parameters
+    are used for logging/early stopping during inference.
+
+    Parameters
+    ----------
+    loss
+        Tensor with loss for minibatch. Should be one dimensional with one value.
+        Note that loss should be in an array/tensory and not a float.
+    reconstruction_loss
+        Reconstruction loss for each observation in the minibatch. If a tensor, converted to
+        a dictionary with key "reconstruction_loss" and value as tensor
+    kl_local
+        KL divergence associated with each observation in the minibatch. If a tensor, converted to
+        a dictionary with key "kl_local" and value as tensor
+    kl_global
+        Global kl divergence term. Should be one dimensional with one value. If a tensor, converted to
+        a dictionary with key "kl_global" and value as tensor
+    extra_metrics
+        Additional metrics can be passed as arrays/tensors or dictionaries of
+        arrays/tensors.
+    """
+
+    loss: LossRecord
+    reconstruction_loss: Optional[LossRecord] = None
+    kl_local: Optional[LossRecord] = None
+    kl_global: Optional[LossRecord] = None
+    extra_metrics: Optional[Dict[str, Tensor]] = field(default_factory=dict)
```

Comment:
```suggestion
    extra_metrics: Dict[str, Tensor] = field(default_factory=dict)
```

Reply:
I would say this is optional because you don't need to specifiy it